setuptools/command/sdist.py

```diff
@@ -200,22 +200,32 @@ def read_manifest(self):
         manifest.close()
 
     def check_license(self):
-        """Checks if license_file' is configured and adds it to
-        'self.filelist' if the value contains a valid path.
+        """Checks if license_file' or 'license_files' is configured and adds any
+        valid paths to 'self.filelist'.
         """
 
+        files = set()
+
         opts = self.distribution.get_option_dict('metadata')
 
         # ignore the source of the value
         _, license_file = opts.get('license_file', (None, None))
 
         if license_file is None:
             log.debug("'license_file' option was not specified")
-            return
+        else:
+            files.add(license_file)
 
-        if not os.path.exists(license_file):
-            log.warn("warning: Failed to find the configured license file '%s'",
-                    license_file)
-            return
+        try:
+            files.update(self.distribution.metadata.license_files)
+        except TypeError:
+            log.warn("warning: 'license_files' option is malformed")
+
+        for f in files:
+            if not os.path.exists(f):
+                log.warn(
+                    "warning: Failed to find the configured license file '%s'",
+                    f)
+                continue
 
-        self.filelist.append(license_file)
+            self.filelist.append(f)
```

Comment:
Use `self.filelist.extend(files)` after filtering out any missing.

Reply:
Is this for efficiency? I ended up going for mutating the set while iterating in e08ec2b640f6b2bf943fea70e2a7f9881bbe6e91, but I'm not sure that's the best. It seemed like appending filtered files to an intermediate list/set before doing the extend would cancel out any performance benefits. Maybe there's a better way to do it?

Reply:
I generally avoid incremental changes to any mutable object, as that leaves the object in lots of intermediate states, so it's more obvious what the end effect is if it happens in a single instruction instead of in a lot of instructions. That said, I'm not staunchly opposed to this approach. Happy to proceed with however this ended up.